### PR TITLE
Fix notes section on rewards does not expand and gets cut off

### LINKED ISSUE #1301: notes section on rewards does not expand and gets cut off
> iOS Version: 16.5
> Device: iPhone 13
> App Version: 3.7.2
> User ID: 14cb6b7c-5af9-4a5c-995a-2d45bdb60c76
>
>Please describe the bug you encountered:
> iOS mobile app only shows the first line of the reward descriptions. full descriptions show up on the mobile and desktop websites tho. 
> IMG_0575
> IMG_0576

### PR DESCRIPTION
**Pull request by giovannapezzini**
Closes #1301 

my Habitica User-ID: 76ec69ee-f02d-4b55-b622-c0d8e381d155

---

Hello 👋 happy to make my first contribution to the project.

Changes:
- Replaces `sizeForItemAt` and `insetForSectionAt` collection view methods for a `UICollectionViewCompositionalLayout `
- Set number of lines of `CustomRewardCell` to zero so it's expandable

> **Note**: 
> When rotating the iPhone I notice this bug in production, this is also fixed in this pull request
> 

#### Demo

| Multiple Lines |
| -------------  |
|<video src="https://github.com/HabitRPG/habitica-ios/assets/38385635/9a5a2fbc-4b76-4d5a-9a88-ff6861577877"> |

| iPhone 8 | iPhone 14 |
| --------- | ---------- |
| <video src="https://github.com/HabitRPG/habitica-ios/assets/38385635/3a26ac38-95c3-497b-bd19-3a1f3187c5de"> | <video src="https://github.com/HabitRPG/habitica-ios/assets/38385635/58fd4395-b42a-447c-b6f3-ec91d573ff6c"> | 

| iPad Mini | iPad Pro |
| --------- | -------- |
|(6th generation)iPad mini (6th generation)|iPad Pro (12 9-inch) (5th generation)iPad Pro (12 9-inch) (5th generation)|

### CODE REVIEW

File: HabitRPG/UI/RewardViewController.swift [hunk 1]
```diff
@@ -10,6 +10,38 @@ import UIKit
 import Habitica_Models
 import ReactiveSwift
 
+enum RewardSectionType {
+    case custom
+    case inApp
```
Comment (giovannapezzini) [inline HabitRPG/UI/RewardViewController.swift:15]: Cases based on the cells name `CustomRewardCell` and `InAppRewardCell`

File: HabitRPG/UI/RewardViewController.swift [hunk 2]
```diff
@@ -134,6 +152,24 @@ class RewardViewController: BaseCollectionViewController, UICollectionViewDelega
...
+        let layout = UICollectionViewCompositionalLayout { [self] sectionIndex, _ in
+            let isCustomRewardsSection = self.dataSource.isCustomRewardsSection(sectionIndex)
+            let sectionType: RewardSectionType = isCustomRewardsSection ? .custom : .inApp
+            let itemSize = NSCollectionLayoutSize(widthDimension: sectionType.itemWidth,
+                                                  heightDimension: sectionType.itemHeight)
+            let item = NSCollectionLayoutItem(layoutSize: itemSize)
+            let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
+                                                   heightDimension: sectionType.itemHeight)
+            let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitems: [item])
+            group.interItemSpacing = .flexible(1)
```
Comment (giovannapezzini) [inline HabitRPG/UI/RewardViewController.swift:166]: This is the minimum spacing between items, we need this so `collectionView.autoresizingMask` will work properly and items will fill the width of the screen, without these, cells would be aligned to the left

**Comment by giovannapezzini** [comment]
This Pull Request is ready for review! I'm used to leave comments to give a bit of context, please let me know you have any other questions or inquiries 🙂 Have a good weekend!

**Comment by saraolson** [comment]
@giovannapezzini thanks so much for your contribution! we'll review it in a test build and let you know if we need any changes. if everything looks good, we'll grant you contributor credit for your habitica profile :)

**Comment by saraolson** [comment]
@giovannapezzini Everything looks good! this PR will be released in the next iOS update, 3.8. I've added this contribution to your habitica profile and granted you a contributor tier 🎉

**Pull request merged**